### PR TITLE
Create e2e job for SDN multitenant mode

### PR DESCRIPTION
**Pull request by rcarrillocruz**
This way we will be able to inject different values for SDN (potentially OVN)
when defining jobs that consume this template

### CODE REVIEW

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml [hunk 1]
```diff
@@ -187,6 +187,8 @@ objects:
       env:
       - name: TYPE
         value: ${CLUSTER_TYPE}
+      - name: NETWORK_TYPE
```
Comment (abhinavdahiya) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: CLUSTER_TYPE is a template variable and therefore probably you can change that. this would need to be something similar.

Comment (rcarrillocruz) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: Sorry but I don't follow what CLUSTER_TYPE has to do here.
I'm adding a new env variable, just so the networkType can be configured, instead of using the current default which is OpenShiftSDN.

CLUSTER_TYPE is a var to signal where should be the cluster be created, whether GCP or AWS.

**Comment by squeed** [comment]
Seems reasonable.

Can you chat with DevEx and figure out the right way for failures to be reported?

**Comment by rcarrillocruz** [comment]
Chatted with DeVex, they say the only way to see failures is to just check Deck.

File: ci-operator/jobs/openshift/origin/openshift-origin-periodics.yaml [hunk 2]
```diff
@@ -62,6 +62,64 @@ periodics:
...
       name: job-definition
+- agent: kubernetes
+  cron: ""
+  decorate: true
+  extra_refs:
+  - base_ref: master
+    org: openshift
+    repo: origin
+  interval: 24h
+  name: periodic-ci-e2e-aws-multitenant
```
Comment (smarterclayton) [inline ci-operator/jobs/openshift/origin/openshift-origin-periodics.yaml]: This should be structured like a release periodic, not a post submit periodic.  We want to run tests on releases, not tests on branches.

Add this to openshift-release-release-4.2-periodics.yaml as `release-....<same_as_others>...-e2e-aws-network-multitenant-4.2`

Comment (smarterclayton) [reply ci-operator/jobs/openshift/origin/openshift-origin-periodics.yaml]: You'll want to set the RELEASE_IMAGE_LATEST to `registry.svc.ci.openshift.org/ocp/release:4.2`

**Comment by squeed** [comment]
@smarterclayton do you have any ideas about how to raise failures for this? It's enough of an obscure use-case that I don't think there's a need to gate CI, but we shouldn't let this be failing for releases.

We have test coverage from QE too.

**Comment by danwinship** [comment]
We need to do something to skip all `[Feature:NetworkPolicy]` tests when running under multitenant.

Test skipping is controlled by code in `origin/test/extended/util/test.go`. It looks like there's no way to specify additional skips via environment variables currently, but maybe that would be the right way to do this.

Another possibility is that there's already code in the `origin/test/extended/networking` tests to have some tests only run with some plugins. (This doesn't apply to the `[Feature:NetworkPolicy]` tests though because they come from upstream.) Maybe instead of having those tests get skipped just-in-time, we could rewrite things so that `testMaps` gets updated at startup time based on the plugin. (eg, if we're testing against ovs-multitenant it would add elements to `testMaps` to disable `[Feature:NetworkPolicy]`, `should allow communication between pods in different namespaces on the same node`, `should allow communication between pods in different namespaces on different nodes`, etc. And if we're testing ovs-networkpolicy, it would disable `should prevent communication between pods in different namespaces on the same node`, etc.)

**Comment by rcarrillocruz** [comment]
Multitenant test suite is on https://github.com/openshift/origin/pull/23321/files

File: ci-operator/jobs/openshift/release/openshift-release-release-4.2-periodics.yaml [hunk 3]
```diff
@@ -123,6 +123,90 @@ periodics:
     - name: pull-secret
       secret:
         secretName: ci-pull-credentials
+- agent: kubernetes
+  cron: '@yearly'
```
Comment (smarterclayton) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.2-periodics.yaml]: You have to set an interval (see how the rollback jobs are set)

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml (hunk 1 above)
Comment (smarterclayton) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: CLUSTER_TYPE is supposed to describe the type of cluster (a profile), not just the platform.  "aws-network-multitenant" would thus be a valid cluster type.

However, at this point you can't easily combine those (aws + multitenant), so this is an acceptable hack.

However, since you don't use NETWORK_TYPE anywhere you shouldn't abstract it.  Please remove this reference and simply leave the network manifest.

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml [hunk 4]
```diff
@@ -602,6 +607,11 @@ objects:
             exit 1
         fi
 
+        if [[ -n "${CLUSTER_NETWORK_MANIFEST}" ]]; then
+            openshift-install --dir=/tmp/artifacts/installer/create manifests
```
Comment (smarterclayton) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: This won't really work once we add more types of injection, and may not be how we decide to do this injection (profiles were supposed to be a config map of files injected into the pod that were then referenced by this job).

Please leave a `TODO: Replace with a more concise manifest injection approach`

**Comment by dcbw** [comment]
@smarterclayton look better now?

File: ci-operator/jobs/openshift/release/openshift-release-release-4.2-periodics.yaml [hunk 5]
```diff
@@ -123,6 +123,90 @@ periodics:
     - name: pull-secret
       secret:
         secretName: ci-pull-credentials
+- agent: kubernetes
+  cron: ""
+  decorate: true
+  interval: 12h
+  labels:
+    job-env: aws
+    job-test: e2e
```
Comment (smarterclayton) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.2-periodics.yaml]: These labels don't mean anything for you, you should remove them.  They are used by other jobs that have additional logic to tie them to cluster bot.  So no labels for this job.

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml [hunk 6]
```diff
@@ -523,6 +523,7 @@ objects:
       env:
       - name: TYPE
         value: ${CLUSTER_TYPE}
+      - name: CLUSTER_NETWORK_MANIFEST
```
Comment (smarterclayton) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: You need to declare this as an optional template parameter or the value won’t get passed down to you.

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml [hunk 7]
```diff
@@ -20,6 +20,9 @@ parameters:
 - name: RELEASE_IMAGE_LATEST
   required: true
 - name: BASE_DOMAIN
+  value: origin-ci-int-aws.dev.rhcloud.com
+  required: true
```
Comment (smarterclayton) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: why did you add this?

Comment (rcarrillocruz) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: must've been a bogus rebase, fixing...

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml [hunk 8]
```diff
@@ -652,6 +655,12 @@ objects:
             exit 1
         fi
 
+        # TODO: Replace with a more concise manifest injection approach
+        if [[ -n "${CLUSTER_NETWORK_MANIFEST}" ]]; then
+            openshift-install --dir=/tmp/artifacts/installer/create manifests
+            echo "$CLUSTER_NETWORK_MANIFEST" > /tmp/artifacts/installer/manifests/cluster-network-03-config.yml
```
Comment (smarterclayton) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: Why is the one above "${CLUSTER_NETWORK_MANIFEST}" and this one "$CLUSTER_NETWORK_MANIFEST"?

If they are both "${CLUSTER_NETWORK_MANIFEST}" you don't need to declare the pod env var on the template.

Comment (rcarrillocruz) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: no particular reason, i just mixed syntax. Since I didn't put a template variable, I did not even think about this. I will amend, and remove the envvar.
